### DISCUSSION TITLE
Razor Page Error Handling instead of Blazor in Blazor Server Project?

### DISCUSSION THREAD

**Discussion by magicjar** [abpframework/abp#21652] [Q&A]
Is it possible to use Razor Page error handling instead of redirecting to Blazor's 'Sorry, there's nothing at this address.' page?

I'm using a Blazor Server project where I can create a Razor Page site with some Blazor interactivity.

Thanks.

Edit:

I'm using Basic Theme.

I have tried adding bellow code in `OnApplicationInitialization` and create 404.cshtml inside Pages/Error

```
app.UseExceptionHandler("/Error"); // For server-side errors
app.UseStatusCodePagesWithReExecute("/Error/{0}"); // Handles 404, etc.
```

but it still did not work and it still redirect to Blazor `<NotFound>` bellow.

```
@using Microsoft.Extensions.Options
@using Volo.Abp.AspNetCore.Components.Web.Theming.Routing
@inject IOptions<AbpRouterOptions> RouterOptions
<CascadingAuthenticationState>
    <Router AppAssembly="RouterOptions.Value.AppAssembly"
            AdditionalAssemblies="RouterOptions.Value.AdditionalAssemblies">
        <Found Context="routeData">
            <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)">
                <NotAuthorized>
                    @if (!context.User.Identity.IsAuthenticated)
                    {
                        <RedirectToLogin />
                    }
                    else
                    {
                        <p>You are not authorized to access this resource.</p>
                    }
                </NotAuthorized>
            </AuthorizeRouteView>
        </Found>
        <NotFound>
            <LayoutView Layout="@typeof(MainLayout)">
                <p>Sorry, there's nothing at this address.</p>
            </LayoutView>
        </NotFound>
    </Router>
</CascadingAuthenticationState>
```

Everything else are untouched.

Created using `abp new Acme.BookStore -u blazor-server -dbms PostgreSQL -m none --theme basic -csf`

**Comment by maliming** [comment]
hi

> Is it possible to use Razor Page error handling instead of redirecting to Blazor's 'Sorry, there's nothing at this address.' page?

Can you share some screenshots?

**Reply by magicjar** [reply]
What screenshot sorry?

I have edited my question above to add some details.

**Reply by maliming** [reply]
hi

You can add a custom component and use it in ` <NotFound>`, in your component redirect to MVC page.

**Reply by magicjar** [reply]
Well, yeah, that's my last solution if there is no other option.
